### PR TITLE
Add createImpersonationToken for User

### PR DESCRIPTION
**Pull request by interinvest**

### CODE REVIEW

**Comment by m1guelpf** [comment]
@interinvest Could you add a test for this?

File: lib/Gitlab/Api/Users.php [hunk 1]
```diff
@@ -235,4 +235,16 @@ public function email($id)
...
     }
+    
+    /**
+    * @param int $user_id
+    * @return mixed
+    */
+    public function createImpersonationToken($user_id)
+    {
+        return $this->post('users/'. $this->encodePath($user_id).'/impersonation_tokens', array(
+            'name'    => 'inter-invest',
```
Comment (fbourigault) [inline lib/Gitlab/Api/Users.php]: The token name must be a method parameter.

File: lib/Gitlab/Api/Users.php [hunk 2]
```diff
@@ -235,4 +235,16 @@ public function email($id)
...
+            'scopes[]'  => 'api'
```
Comment (fbourigault) [inline lib/Gitlab/Api/Users.php]: Could you make the scope configurable?

**Comment by interinvest** [comment]
I'm sorry, I didn't use to do phpUnitTest

**Comment by m4tthumphrey** [comment]
Those single quotes look like backticks to me...

**Comment by marylly** [comment]
+1

**Comment by GrahamCampbell** [comment]
Closing due to inactivity. If anyone would still like this, please submit a new PR with tests.

**Pull request closed without merging**